### PR TITLE
🦋 Allow to add a specific meta balise on CMS page #1470

### PR DESCRIPTION
**Pull request by ithiame**

### CODE REVIEW

**Review by coyotte508 — APPROVED** [review]

**Comment by ithiame** [comment]
image

image

**Comment by Tirodem** [comment]
image
CTA should be "Add custom meta balise", not "Add variation"

File: src/routes/(app)/admin[[hash=admin_hash]]/cms/new/+page.server.ts [hunk 1]
```diff
@@ -28,9 +36,14 @@ export const actions = {
 				fullScreen: z.boolean({ coerce: true }),
 				maintenanceDisplay: z.boolean({ coerce: true }),
 				hasMobileContent: z.boolean({ coerce: true }),
-				mobileContent: z.string().max(MAX_CONTENT_LIMIT).optional()
+				mobileContent: z.string().max(MAX_CONTENT_LIMIT).optional(),
+				hasCustomMeta: z.boolean({ coerce: true }),
```
Comment (coyotte508) [inline src/routes/(app)/admin[[hash=admin_hash]]/cms/new/+page.server.ts:40]: You should just ignore this parameter and base everything on `metas.length`.

Same in the update logic.

Currently I think if I remove all the metas of the CMS page in the UI it maybe doesn't work?

File: src/lib/components/CmsForm.svelte [hunk 2]
```diff
@@ -26,14 +30,16 @@
...
-
+	let hasCustomMeta = !!cmsPage?.metas?.length;
 	let hasMobileContent = cmsPage?.hasMobileContent || false;
 	let mobileContent = cmsPage?.mobileContent || '';
 	function confirmDelete(event: Event) {
 		if (!confirm('Would you like to delete this CMS page?')) {
 			event.preventDefault();
 		}
 	}
+	let metas = cmsPage?.metas || undefined;
```
Comment (coyotte508) [inline src/lib/components/CmsForm.svelte:41]:
```suggestion
	let metas = cmsPage?.metas;
```

I don't understand what the ` || undefined` does

File: src/lib/components/CmsForm.svelte [hunk 3]
```diff
@@ -26,14 +30,16 @@
...
+	let cmsMetaLine = cmsPage?.metas?.length || 2;
```
Comment (coyotte508) [inline src/lib/components/CmsForm.svelte:42]:
```suggestion
	let cmsMetaLine = cmsPage?.metas?.length ?? 2;
```

File: src/lib/components/CmsForm.svelte [hunk 4]
```diff
@@ -94,7 +100,54 @@
...
+		<input
+			type="checkbox"
+			name="hasCustomMeta"
+			bind:checked={hasCustomMeta}
+			class="form-checkbox"
+		/>
+		Add custom meta tag
+	</label>
+	{#if hasCustomMeta}
+		{#each [...(metas || []), ...Array(cmsMetaLine).fill( { name: '', content: '' } )].slice(0, cmsMetaLine) as meta, i}
```
Comment (coyotte508) [inline src/lib/components/CmsForm.svelte:113]:
```suggestion
		{#each [...(metas ?? []), ...Array(cmsMetaLine).fill( { name: '', content: '' } )].slice(0, cmsMetaLine) as meta, i}
```

**Review by coyotte508 — APPROVED** [review]

**Pull request merged**